sdks/java/io/fileschematransform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileWriteSchemaTransformFormatProviders.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.fileschematransform;
+
+import com.google.auto.service.AutoService;
+import java.util.Map;
+import org.apache.beam.sdk.annotations.Internal;
+import org.apache.beam.sdk.schemas.io.Providers;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+
+/**
+ * {@link FileWriteSchemaTransformFormatProviders} contains {@link
+ * FileWriteSchemaTransformFormatProvider} implementations.
+ *
+ * <p>The design goals of this class are to enable clean {@link
+ * FileWriteSchemaTransformConfiguration#getFormat()} lookups that map to the appropriate {@link
+ * org.apache.beam.sdk.io.FileIO.Write} that encodes the file data into the configured format.
+ */
+@Internal
+public final class FileWriteSchemaTransformFormatProviders {
+  private static final String AVRO = "avro";
+  private static final String CSV = "csv";
+  private static final String JSON = "json";
+  private static final String PARQUET = "parquet";
+  private static final String XML = "xml";
```

Comment:
are these meant to be URNs for the transforms? I believe the format that we arrived after discussing with @ahmedabu98 was `beam:schematransform:org.apache.beam:pubsublite_write:v1` - for this case, i suppose it could be `beam:schematransform:org.apache.beam:files_write:avro:v1` or something like that?

Reply:
No they are allowed inputs of the `setFormat` parameter.  The URN does not drive the format it drives which SchemaTransformProvider.

Reply:
The PR contains a `FileWriteSchemaTransformFormatProvider` interface that extends `Providers.Identify` so that we can do this:
```java
Map<String, FileWriteSchemaTransformFormatProvider> formatProviders = FileWriteSchemaTransformFormatProviders.loadProviders();
if (!formatProviders.containsKey(configuration.getFormat()) {
    throw new IllegalArgumentException(String.format("configuration format is not supported: %s", configuration.getFormat()));
}
FileWriteSchemaTransformFormatProvider provider = providers.get(configuration.getFormat()).buildTransform();

PTransform<PCollection<Row>, PDone> transform = provider.buildTransform();
```

Reply:
sgtm